app/controllers/concerns/pages.rb

```diff
@@ -0,0 +1,39 @@
+module Pages
+  extend ActiveSupport::Concern
+
+  included do
+    resource_actions :default
+
+    schema_type :strong_params
+
+    allowed_params :create, :url_key, :title, :content, :language
+    allowed_params :update, :url_key, :title, :content, :language
+
+    before_filter :set_language_from_header, only: [:index]
+
+    define_method(:link_header) do |resource|
```

Comment:
OOI - why not just extract the instance method def outside the included block with the same protected?